### DISCUSSION TITLE
MiEx: how to use

### DISCUSSION THREAD

**Discussion by ghost** [BramStoutProductions/MiEx#27] [Q&A]
How do u open this up and use it, im super confused

**Comment by BramStout** [comment]
Hi! You can download the latest version of MIEx right here: https://github.com/BramStoutProductions/MiEx/releases It's the MiEx.jar file. You place this file somewhere on your computer and then, assuming that you have Java installed, double click the jar file to open it up.

To learn more about how MiEx actually works, you can read the [Wiki](https://github.com/BramStoutProductions/MiEx/wiki/1.-Home)

**Reply by Lodemor** [reply]
Hi, I'm having a similar problem, MiEX.jar just won't run on my system it seems (windows 10). I've made sure that java is working, other jar executables like optifine do open up when i run them, just seems miex that is refusing to open.

Something I'd like to note is that previous versions work (v1.1.0 is the latest that works, all newer ones do not open)

I have reinstalled java multiple times as well incase it was a java problem but still nothing. Do you have any other troubleshooting ideas?

**Reply by BramStout** [reply]
Does MiEx generate a `log.txt` file next to the jar file when you try to run it? If so, could you share it here? It most likely will say what's going on.

If there isn't a `log.txt` file, what you could then do is open up the folder containing the MiEx.jar file in File Explorer, shift right click on an empty area, and hit the open command prompt or open powershell option. In the command prompt or powershell type the following command `java -jar MiEx.jar` (assuming that the MiEx jar file is called `MiEx.jar`). If it tells you that java isn't a recognized program, then try the following command `&"C:\Program Files\Java\jre-1.8\bin\java.exe" -jar MiEx.jar` (of course the file path to `java.exe` that's in between double quotes should be replaced with the file path to where Java is installed on your computer). 

If there is some issue, it should print the error out in there. You can then share it here and I can help you fix it.

**Reply by Lodemor** [reply]
OKAY, i did it the minecraft server style, where i made a .bat file to run it with the command you provided, (cuz i didn't see the command prompt option in file explorer) and it now opens up and makes a bunch of folders!

Double clicking still doesnt work tho, but essentially the problem is now fixed, thank you very much!

Since it opened properly, doesn't seem like the log file would show any useful error messages, but just incase:

MiEx v1.4.0
java.runtime.name = Java(TM) SE Runtime Environment
java.vm.name = Java HotSpot(TM) Client VM
sun.os.patch.level = 
java.runtime.version = 1.8.0_411-b09
os.arch = x86
os.name = Windows 10
java.specification.name = Java Platform API Specification
os.version = 10.0
java.specification.version = 1.8
java.specification.vendor = Oracle Corporation
java.version = 1.8.0_411
java.specification.maintenance.version = 5

Installing base_resource_pack
Installing base_resource_pack/miex_config.json
Installing base_resource_pack/materials/minecraft/templates/base.json
Installing base_resource_pack/materials/minecraft/templates/emission.json
Installing base_resource_pack/materials/minecraft/templates/grass_block_side.json
Installing base_resource_pack/materials/minecraft/templates/grass_block_snow.json
Installing base_resource_pack/materials/minecraft/templates/redstone_wire.json
Installing UsdPreviewSurface/materials/minecraft/templates/base.json

**Reply by BramStout** [reply]
It's at least nice to hear that it starts with a .bat script! I think that it not running when double clicking is probably a Windows thing. The only potential issue that I'm noticing when looking at the log, is that you are using a 32-bit version of Java. While it will work with this, it will limit MiEx to a maximum of 4GB of memory, which could cause big slowdowns and maybe an out-of-memory exception in some cases if you're trying to view the world very zoomed out. So, I would recommend to make sure that you're using the 64-bit version of Java.

**Reply by Lodemor** [reply]
understood!

**Comment by ghost** [comment]
I have java installed but it opens up the whole file instead of the program

**Comment by BramStout** [comment]
> I have java installed but it opens up the whole file instead of the program
> […](#)
> On Thu, Apr 25, 2024 at 4:23 AM Bram Stout ***@***.***> wrote: Hi! You can download the latest version of MIEx right here: https://github.com/BramStoutProductions/MiEx/releases It's the MiEx.jar file. You place this file somewhere on your computer and then, assuming that you have Java installed, double click the jar file to open it up. To learn more about how MiEx actually works, you can read the Wiki <https://github.com/BramStoutProductions/MiEx/wiki/1.-Home> — Reply to this email directly, view it on GitHub <[#27 (comment)](https://github.com/BramStoutProductions/MiEx/discussions/27#discussioncomment-9222254)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BICEMO3XTJ47NN62CS7TFYDY7C4RLAVCNFSM6AAAAABGYAPS5SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TEMRSGI2TI> . You are receiving this because you authored the thread.Message ID: ***@***.*** com>

Windows then probably wasn't told to run Jar files but instead open them up as zip files. If you right click them, you can hit "Open With" and from there select Java or you might have to select "Choose Another App" and then select Java from there.

**Reply by ghost** [reply]
It says unable to open in Java

**Reply by BramStout** [reply]
Weird. Would you be able to share a screenshot or screen-recording of the issue? That would make it a lot easier for me to see what's going on.

**Reply by ghost** [reply]
All i did is click open in Java and either nothing happens or it says error. Maybe I dont have the right java

**Reply by BramStout** [reply]
In theory the latest 64 bit Java 8 or later should work. You can download Java 8 here: https://www.java.com/en/download/manual.jsp

**Reply by ghost** [reply]
now it just opens and quickly disappears

**Reply by ghost** [reply]
I fixed it!

**Reply by BramStout** [reply]
Great! Is everything working now?

**Reply by ghost** [reply]
It seems like everything is working, I dont see a resources folder though.

**Reply by ghost** [reply]
How do u add your own custom texture pack? Does it extract it from the world

**Reply by BramStout** [reply]
The `resources` folder is located in the "current working directory" when you launch MiEx. If you launch MiEx by double clicking on it, then the current working directory is the folder where the jar file is located, so the `resources` folder is right next to the MiEx.jar file. If you launch it via a .bat file, then the current working directory is the folder that the .bat file is in, so the `resources` folder would be next to the .bat file. If you launch MiEx via the command line, then the current working directory is noted before the `>` where you type in the command. You can also manually say where the resources folder should be ( https://github.com/BramStoutProductions/MiEx/wiki/3.-Pipeline-Integration-and-Customisations#environment-variables )

To add in custom resource packs, you place the resource packs in the `resources` folder. MiEx only supports resource packs in the form of folders and not zip files. So if it's as a zip file, you'll have to extract it into a folder first. MiEx doesn't extract the resource packs from the world.
